### PR TITLE
many: move to purely hash based key lookup and to new key/signature format (v1)

### PR DESCRIPTION
**Pull request by pedronis**
blocked on:

* still some failing tests
* really need new format trusted assertions for asserts/sysdb,  debian/rules will also need corresponding updates

### CODE REVIEW

File: asserts/crypto.go [hunk 1]
```diff
@@ -104,54 +108,43 @@ func signContent(content []byte, privateKey PrivateKey) ([]byte, error) {
...
+		return nil, fmt.Errorf("unsupported %s format version: %d", kind, buf[0])
 	}
-	pkt, err := packet.Read(bytes.NewReader(data))
+	rd := bytes.NewReader(buf[1:])
+	pkt, err := packet.Read(rd)
 	if err != nil {
 		return nil, fmt.Errorf("cannot decode %s data: %v", kind, err)
 	}
+	if rd.Len() != 0 {
+		return nil, fmt.Errorf("%s has spurious trailer data", kind)
```
Comment (emgee) [inline asserts/crypto.go]: s/trailer/trailing

File: asserts/account_key.go [hunk 2]
```diff
@@ -48,14 +48,14 @@ func (ak *AccountKey) Until() time.Time {
...
+// PublicKeyID returns the key id used for lookup of the account key.
 func (ak *AccountKey) PublicKeyID() string {
 	return ak.pubKey.ID()
 }
 
-// PublicKeyFingerprint returns the fingerprint of the account key.
-func (ak *AccountKey) PublicKeyFingerprint() string {
-	return ak.pubKey.Fingerprint()
+// PublicKeySHA3_384 returns the sha3-384 hash used as id for lookup of the account key, same as PublicKeyID.
+func (ak *AccountKey) PublicKeySHA3_384() string {
```
Comment (niemeyer) [inline asserts/account_key.go]: Can we please kill that?  We have PublicKeyID already.

File: asserts/asserts.go [hunk 3]
```diff
@@ -97,8 +98,8 @@ type Assertion interface {
 	// Signature returns the signed content and its unprocessed signature
 	Signature() (content, signature []byte)
 
-	// SigningKey returns the signer identifier and key id for the key that signed this assertion.
-	SigningKey() (signerID, keyID string, err error)
+	// SigningKey returns the key id for the key that signed this assertion.
+	SigningKey() string
```
Comment (niemeyer) [inline asserts/asserts.go]: Can we rename this to SignKey, following the new header name?

Comment (niemeyer) [inline asserts/asserts.go]: SignKeyID actually, conforming to the use on PublicKeyID.

File: asserts/crypto.go [hunk 4]
```diff
@@ -104,134 +108,112 @@ func signContent(content []byte, privateKey PrivateKey) ([]byte, error) {
...
+	buf := make([]byte, base64.StdEncoding.DecodedLen(len(b)))
+	n, err := base64.StdEncoding.Decode(buf, b)
 	if err != nil {
-		return nil, fmt.Errorf("%s: %v", kind, err)
+		return nil, fmt.Errorf("%s: cannot decode base64 data: %v", kind, err)
 	}
-	if format != "openpgp" {
-		return nil, fmt.Errorf("unsupported %s format: %q", kind, format)
+	if n == 0 {
+		return nil, fmt.Errorf("empty decoded %s", kind)
```
Comment (niemeyer) [inline asserts/crypto.go]: "empty decoded" reads a bit strange. Maybe "cannot decode %q: base64 without data"?

Comment (niemeyer) [inline asserts/crypto.go]: Similarly, "cannot decode %q: no data" above, instead of "empty %s"?

**Comment by niemeyer** [comment]
A few more simple comments, and LGTM.

**Pull request merged**